### PR TITLE
Feature/add confirmed lesson screen

### LINKED ISSUE #64: Add the edit Scheduled Lesson screen

### PR DESCRIPTION
**Pull request by Philipp0206**
* Create the `ConfirmedLessonScreen` screen, add it in the navigation graph
  The confirmed lesson screen displays the lesson, its location, and associated tutor. It can be accessed from the overview screen.
* Add tests for the `EditRequestedLessonScreen`
  Add tests for previously implemented screen, as decided with the team (see #72 for screen implementation)

> [!NOTE] 
> The tests for the `ConfirmedLessonScreen` will be added in a following pull request.
> @AlexisAllmnd slightly modified the UI and added the message redirection.

Close #64

### CODE REVIEW

File: app/src/main/java/com/github/se/project/ui/lesson/ConfirmedLesson.kt [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+              }
+            }
+      },
+      content = { paddingValues ->
+        Column(
+            modifier = Modifier.fillMaxSize().padding(horizontal = 24.dp).padding(paddingValues),
+            verticalArrangement = Arrangement.spacedBy(16.dp)) {
+              // Use DisplayLessonDetails to show lesson and tutor details
+              DisplayLessonDetails(
+                  lesson = lesson, studentProfile = tutorProfile // Passing the tutor's profile
```
Comment (AlexisAllmnd) [inline app/src/main/java/com/github/se/project/ui/lesson/ConfirmedLesson.kt:52]: Here, you should adapt this depending who is the current user. If the user is a tutor, display student info. Else display tutor.

**Review by AlexisAllmnd — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/github/se/project/ui/lesson/ConfirmedLesson.kt [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+  val lesson =
+      lessonViewModel.selectedLesson.collectAsState().value
+          ?: return Text("No lesson selected. Should not happen.")
+
+  val otherProfile =
+      if (isStudent) {
+        listProfilesViewModel.getProfileById(lesson.tutorUid[0])
+      } else {
+        listProfilesViewModel.getProfileById(lesson.studentUid)
+      } ?: return Text("Cannot retrieve profile")
```
Comment (AlexisAllmnd) [inline app/src/main/java/com/github/se/project/ui/lesson/ConfirmedLesson.kt:50]: Good change to adapt who to display in function of roles

**Review by AlexisAllmnd — APPROVED** [review]

**Review by AlexisAllmnd — COMMENTED** [review]
I reviewed my point in your last commit but as I modified the Ui and a bit of the way your screen work so I need another reviewer.

File: app/src/main/java/com/github/se/project/MainActivity.kt [hunk 3]
```diff
@@ -103,11 +104,6 @@ fun PocketTutorApp(
               })
         }
       }
-      // For debugging purposes (when sign-in error)
-      // composable(Screen.AUTH) {
-      //   googleUid = "1234"
-      //   CreateProfileScreen(navigationActions, listProfilesViewModel, googleUid)
-      // }
```
Comment (alexs-rufer) [inline app/src/main/java/com/github/se/project/MainActivity.kt:110]: great idea to remove those useless commented lines :)

File: app/src/main/java/com/github/se/project/ui/lesson/ConfirmedLesson.kt [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+              // Lesson Details Card
+              Card(
+                  modifier = Modifier.fillMaxWidth(),
+                  colors =
+                      CardDefaults.cardColors(
+                          containerColor = MaterialTheme.colorScheme.surfaceVariant)) {
+                    Column(
+                        modifier = Modifier.padding(8.dp),
+                        verticalArrangement = Arrangement.spacedBy(8.dp)) {
+                          DisplayLessonDetails(lesson, otherProfile)
```
Comment (alexs-rufer) [inline app/src/main/java/com/github/se/project/ui/lesson/ConfirmedLesson.kt:88]: lgtm

File: app/src/main/java/com/github/se/project/ui/lesson/ConfirmedLesson.kt [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+                        contentDescription = null,
+                        modifier = Modifier.size(20.dp))
+                    Spacer(modifier = Modifier.width(8.dp))
+                    Text("Message ${if (isStudent) "Tutor" else "Student"}")
+                  }
+            }
+      }
+}
+
+private fun formatDateTime(timeSlot: String): String {
```
Comment (alexs-rufer) [inline app/src/main/java/com/github/se/project/ui/lesson/ConfirmedLesson.kt:125]: I think this is a function that we have rewritten 1 or 2 times already, we should extract it in a specific file :) but this is not urgent.

File: app/src/main/java/com/github/se/project/ui/navigation/NavigationActions.kt [hunk 6]
```diff
@@ -30,7 +30,7 @@ object Screen {
   const val ADD_LESSON = "Add Lesson Screen"
   const val TUTOR_MATCH = "Tutor matching Screen"
   const val EDIT_REQUESTED_LESSON = "Edit Requested Lesson"
-  const val EDIT_SCHEDULED_LESSON = "Edit Scheduled Lesson"
+  const val CONFIRMED_LESSON = "Confirmed Lesson"
```
Comment (alexs-rufer) [inline app/src/main/java/com/github/se/project/ui/navigation/NavigationActions.kt:33]: consider renaming it "Confirmed Lesson Screen" for consistency with other screens naming

File: app/src/main/java/com/github/se/project/ui/overview/Overview.kt [hunk 7]
```diff
@@ -61,16 +61,16 @@ fun HomeScreen(
...
+        navigationActions.navigateTo(Screen.CONFIRMED_LESSON)
       }
     } else {
       if (lesson.status == LessonStatus.PENDING_TUTOR_CONFIRMATION) {
         lessonViewModel.selectLesson(lesson)
         navigationActions.navigateTo(Screen.TUTOR_LESSON_RESPONSE)
-      } else { // State STUDENT_REQUESTED or CONFIRMED
-        // lessonViewModel.selectLesson(lesson)
-        // TODO : navigationActions.navigateTo(Screen : LessonInfo )
+      } else if (lesson.status == LessonStatus.CONFIRMED) {
```
Comment (alexs-rufer) [inline app/src/main/java/com/github/se/project/ui/overview/Overview.kt:71]: great job

**Review by alexs-rufer — APPROVED** [review]
The screen looks really cool, great job here! I suggested some small improvements, feel free to implement them. The messaging feature is also great, great work from @AlexisAllmnd too!

File: app/src/main/java/com/github/se/project/ui/navigation/NavigationActions.kt (hunk 6 above)
Comment (Philipp0206) [reply app/src/main/java/com/github/se/project/ui/navigation/NavigationActions.kt]: Good catch, I forgot it.

**Pull request merged**